homeassistant/components/vizio/media_player.py

```diff
@@ -346,8 +341,11 @@ def source_list(self) -> List[str]:
 
     @property
     def app_id(self) -> Optional[str]:
-        """Return the current app."""
-        return self._current_app
+        """Return the ID of the current app."""
+        if self._current_app_config:
+            return vars(self._current_app_config)
```

Comment:
Why do you use `vars` here ?

Reply:
using it instead of `__dict__` because the variable is a custom object. I can change it to that if you'd prefer, just felt cleaner

Reply:
The comment states it returns the current ID of the app, bu that doesn't seem to be the case ? 

Just taking all instance variables is not good, it means we can start pulling in extra info, including things that are not JSON serializable and crash Home Assistant if the object gets updated.

Reply:
Dicts in attributes are rendered very poorly too. What is it that you're trying to achieve ?

Reply:
the ID of the app is a combination of multiple properties which seemed easiest to display as a `dict`. I can display it in the custom object form if you would like, the `__repr__` returns a single line string which is `AppConfig({dict})`

Reply:
Can you give an example ? And do people define it in their config?

btw, I was reading the docs. It's confusing! We should rewrite it and just put on top: Use the config flow. Let's not scare people with manual installing things.

Reply:
sure, for Netflix, you would see:
```json
{
   "NAME_SPACE": 3,
   "APP_ID": "1",
   "MESSAGE": None
}
```
If it's an app that `pyvizio` doesn't recognize, then yes users could add it to their config like so:
```yaml
vizio:
    apps:
        additional_configs:
          - name: Netflix
            config:
                NAME_SPACE: 3
                APP_ID: "1"
                MESSAGE: null
```

Re: docs that's fair, I will do a refactor soon and get rid of the manual pairing instructions. I kept them in there because I don't have a Vizio TV to test this on and wanted to make sure new users weren't completely blocked from adding Vizio TV's once the config flow functionality was released, but it's been out long enough that I feel more comfortable making that the only instructions

Reply:
Just to add, that object will always only have those three attributes because the object is defined solely for the purpose of retrieving the attributes that the SmartCast API provides. I couldn't find a single ID attribute that can be used with the API to identify and launch an app

Reply:
So this seems like an ID that we should only include if the current app is not recognized. I'm fine with keeping it a dict but create your own `{ "id": … }`, don't create a helper.

Reply:
I can make a change so that the ID only shows if the app was unrecognized. That's actually how I first implemented it, just wanted to simplify the code which is why I removed it.

I will also update the code to explicitly pull the attributes I am expecting

Reply:
Just to restate I know my tests are messy and this makes it messier. I will fix it in a future PR

Reply:
@balloob  does this work? want to get this PR approved so I can get started on the test cleanup. I have one more PR to be reviewed before I get there (#33200)